c#-selenium-webdriver-sample/Webdriver.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using OpenQA.Selenium;
+
+namespace CSharpSeleniumWebdriverSample
+{
+    public interface IWebdriver
```

Comment:
I was pretty confused by this interface differing only in 1 character of casing with the real `IWebDriver` interface. Why is this necessary, rather than just returning `IWebDriver`s directly from a WebDriverFactory?

Reply:
(I don't think the individual ChromeWebdriver or FirefoxWebdriver wrappers are necessary, either; I think the extra complexity of splitting the initialization across several files detracts from their use in a sample. Frankly, even the separate `WebdriverFactory` seems like a bit much for a sample; I think a helper method in the actual test class would be fine.

Reply:
this was the first approach that we discussed to support multiple browsers, plus I don't think it's that complex, it's a known pattern that should be easy to follow, still we can change it if you believe a helper function would be better.

Reply:
You can support multiple browsers with the DataRows of a BrowserType enum (this is as far as I remember us discussing) without using a doubly indirected FactoryFactory pattern :)

What we saw in user studies was that, for this sort of sample code, there is a significant number of users who will try to consume the sample by clicking around basically at random between files/links, and will rapidly abandon reading as soon as they start feeling confused or overwhelmed. Because of that, I think it's important to minimize the number of helper files in the sample; it's really important that the structure of the project lead people towards the important entry points.

I'm okay with keeping the separate `WebDriverFactory` static class if you feel it's more clear than making it a helper method in the test class, but I really think the `IWebDriverCreator`, `FirefoxWebDriverCreator`, and `ChromeWebDriverCreator` are adding an extra layer of abstraction for no benefit; if you were to inline the browser creators into the `WebDriverFactory`, it would actually make the factory **shorter**:

```cs
public static IWebDriver GetWebDriver(BrowserType browser)
{
    switch (browser)
    {
        case BrowserType.Chrome:
            var chromeDriverDirectory = Environment.GetEnvironmentVariable("ChromeWebDriver") ?? Environment.CurrentDirectory;
            return new ChromeDriver(chromeDriverDirectory);

        case BrowserType.Firefox:
            var geckoDriverDirectory = Environment.GetEnvironmentVariable("GeckoWebDriver") ?? Environment.CurrentDirectory;
            return new FirefoxDriver(geckoDriverDirectory);
    }
}
```